dspace-uclouvain/src/main/java/org/dspace/uclouvain/submissionMetadataGenerators/consumer/LicenseConsumer.java

```diff
@@ -0,0 +1,46 @@
+package org.dspace.uclouvain.submissionMetadataGenerators.consumer;
+
+import org.dspace.content.Bitstream;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.event.Consumer;
+import org.dspace.event.Event;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+/**
+ * Consumer used to add a default license to a bitstream that does not have one.
+ * Only adds one if the metadata field is empty.
+ * 
+ * Authored-By: Michaël Pourbaix <michael.pourbaix@uclouvain.be>
+ */
+public class LicenseConsumer implements Consumer {
+
+    String LICENSE_URL = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("uclouvain.bitstream.default.license", "https://creativecommons.org/licenses/by/4.0/");
+
+    BitstreamService bitstreamService = ContentServiceFactory.getInstance().getBitstreamService();
+
+    @Override
+    public void finish(Context context) throws Exception {
+    }
+
+    @Override
+    public void initialize() throws Exception {
+    }
+
+    @Override
+    public void consume(Context context, Event event) throws Exception {
+        // Only process bitstreams that have no license && are being modified
+        if (event.getSubjectType() == Constants.BITSTREAM && (event.getEventType() == Event.MODIFY_METADATA || event.getEventType() == Event.MODIFY)) {
+            Bitstream bitstream = (Bitstream) event.getSubject(context);
+            if (bitstreamService.getMetadataFirstValue(bitstream, "dc", "rights", "license", null) == null) {
```

Comment:
Here you can use `getMetadataFirstValue(T, MetadataFieldName, language)` method signature.
The `MetadataFieldName` could be a bean or could be create into the class constructor based on property from configuration. 
I used the second option in my PR.... because i'm not yet fluent with bean ; but if multiple classes use same logic, it could make sense.

```java
MetadataFieldName accessMetadataFieldName;
...
constructorMethod() {
    String fieldName = configurationService.getProperty("global.license.metadata.field", "dc.rights.license");
    this.accessMetadataFieldName = new MetadataFieldName(fieldName);
}
```